### PR TITLE
Include index model degraded status as stats for given node

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Add model_indexing_degraded stat to node stats response.
 
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/knn/plugin/stats/KNNStatsConfig.java [hunk 1]
```diff
@@ -81,5 +86,10 @@
                     new KNNCounterSupplier(KNNCounter.SCRIPT_QUERY_REQUESTS)))
             .put(StatNames.SCRIPT_QUERY_ERRORS.getName(), new KNNStat<>(false,
                     new KNNCounterSupplier(KNNCounter.SCRIPT_QUERY_ERRORS)))
+            .put(StatNames.MODEL_INDEXING_DEGRADED.getName(), new KNNStat<>(false,
+                    new EventOccurredWithinThresholdSupplier(
```
Comment (jmazanec15) [inline src/main/java/org/opensearch/knn/plugin/stats/KNNStatsConfig.java:90]: Good name choice. I wasnt sure what to call this, but this looks good.

File: src/main/java/org/opensearch/knn/plugin/stats/StatNames.java [hunk 2]
```diff
@@ -44,6 +44,7 @@
     INDICES_IN_CACHE("indices_in_cache"),
     CIRCUIT_BREAKER_TRIGGERED("circuit_breaker_triggered"),
     MODEL_INDEX_STATUS("model_index_status"),
+    MODEL_INDEXING_DEGRADED("model_indexing_degraded"),
```
Comment (jmazanec15) [inline src/main/java/org/opensearch/knn/plugin/stats/StatNames.java:47]: I think I rather call this index_from_model_degraded. model_indexing_degraded may be confused with indexing a model is degraded.

Comment (VijayanB) [reply src/main/java/org/opensearch/knn/plugin/stats/StatNames.java]: Ack

**Review by jmazanec15 — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM ! Thanks

**Pull request merged**